enterprise/server/backends/authdb/authdb.go

```diff
@@ -971,6 +978,26 @@ func (d *AuthDB) GetUserAPIKeys(ctx context.Context, groupID string) ([]*tables.
 		return nil, err
 	}
 
+	// If trying to access keys for another user, ORG_ADMIN capability is
+	// required, and the user must be a member of the org.
+	if userID != u.GetUserID() {
+		caps, err := capabilities.ForAuthenticatedUserGroup(ctx, d.env, groupID)
+		if err != nil {
+			return nil, err
+		}
+		if !slices.Contains(caps, akpb.ApiKey_ORG_ADMIN_CAPABILITY) {
+			return nil, status.PermissionDeniedError("missing required capability")
+		}
+		ok, err := d.isGroupMember(ctx, groupID, userID)
```

Comment:
Is this redundant with `ForAuthenticatedUserGroup` which also checks group membership?

Reply:
`ForAuthenticatedUserGroup` uses the authenticated user ID rather than the requested `user_id` which is different in this case